### PR TITLE
Stop pinning `grpcio`

### PR DESCRIPTION
**Pull request by j-bennet**
See if this works now.

### CODE REVIEW

File: ci/environment-3.10.yaml [hunk 1]
```diff
@@ -10,6 +10,6 @@ dependencies:
   - pyarrow
   - pytest
   - gcsfs
-  - grpcio<1.45
```
Comment (ncclementi) [inline ci/environment-3.10.yaml:13]: If things pass, I wonder if we need it as a explicit dependency, I can't recall exactly but I have the hunch that `grpcio` gets installed with google-cloud-bigquery, in fact I think it is see https://github.com/googleapis/python-bigquery/blob/main/setup.py

We might be able to remove it all the way if we do not need the pin.

Comment (j-bennet) [reply ci/environment-3.10.yaml:13]: 👍

**Comment by ncclementi** [comment]
I re-trigger the macos jobs because they timed out, not sure why maybe the availability of instances. Will see what happens once it's done

**Comment by bnaul** [comment]
Ok by me to go ahead and merge given that the only straggler is a Mac build

**Comment by ncclementi** [comment]
+1. to merge, not sure why macos and python 3.9 always hang.

**Pull request merged**